src/App.tsx

```diff
@@ -1,14 +1,64 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+const filterTodos = (todos: Todo[], selectedOption: string, searchQuery: string) => {
+  let preparedTodos = [...todos];
+
+  if (searchQuery) {
+    const preparedSearchQuery = searchQuery.toLowerCase().trim();
+
+    preparedTodos = preparedTodos.filter(todo => todo.title.toLowerCase().includes(preparedSearchQuery));
+  }
+
+  if (selectedOption === 'active') {
+    preparedTodos = preparedTodos.filter(todo => !todo.completed);
+  }
+
+  if (selectedOption === 'completed') {
+    preparedTodos = preparedTodos.filter(todo => todo.completed);
+  }
+
+  return preparedTodos;
+};
```

Comment:
1. Let's move it to separate file `helpers`
2. Let's use 1 filter
3. Change it to switch case
4. Let's create enum FilterType with all filter types

```suggestion
const filterTodos = (todos: Todo[], selectedOption: string, searchQuery: string) => {
  let preparedTodos = [...todos];
  const preparedQuery = searchQuery.toLowerCase().trim();

  return preparedTodo.filter(todo => {
     const isTodoIncluded = todo.title.toLowerCase().includes(preparedQuery);
     
     switch(selectedOption) {
       case FilterType.All:
         return isTodoIncluded;

       case FilterType.Active:
         ....
     }
  }

```